CMakeLists.txt

```diff
@@ -0,0 +1,105 @@
+# 1. Requirements
+cmake_minimum_required(VERSION 3.21)
+
+project(
+OpenAssetIO-MediaCreation
+VERSION 1.0.0
+)
+
```

Comment:
[fix] Can we document here the reasons for our strategy for when we invoke traitgen (compile vs build) and the learnings went through to get to this point. Thats the critical stuff to track. It could go in the commit message, but comments here are probably the most useful long-term.

Reply:
https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/25/commits/74b87cd860c2eb786a1e41e52a56c2c2d4c5c33e

Added a doc explainer.

Reply:
It would be good to also capture [this](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/25#issuecomment-1462289420) as otherwise it sounds like a purely philosophical point, rather than 'its actually not really doable in CMake without a lot of cruft'. Then hopefully we dont get into random third parties questioning it purely on conceptual grounds and the associated churn there.

Reply:
I'll admit I thought I did capture the important parts of the comment. I'm assuming you meant the alternate considerations arn't expounded on? Added them here : https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/25/commits/e100cd9809c5d90f58d3221ff26fd02e61edb02d